metadata-io/src/main/java/com/linkedin/metadata/system_info/SystemInfoService.java

```diff
@@ -0,0 +1,142 @@
+ *   <li>Sensitive properties (passwords, secrets, keys) are automatically redacted
+ *   <li>Property filtering is applied to prevent accidental exposure of credentials
+ * </ul>
+ *
+ * <p><strong>Performance:</strong>
+ *
+ * <ul>
+ *   <li>Uses parallel execution for improved performance
+ *   <li>Includes timeouts for remote component fetching
+ *   <li>Graceful degradation when components are unavailable
+ * </ul>
+ *
+ * @see SystemInfoController for REST API endpoints
+ * @see PropertiesCollector for configuration property collection
+ * @see SpringComponentsCollector for component status collection
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SystemInfoService {
+
+  // Thread pool for parallel execution
+  private final ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+  // Collectors
+  private final SpringComponentsCollector springComponentsCollector;
+  private final PropertiesCollector propertiesCollector;
+
+  /**
+   * Get Spring components information in parallel.
+   *
+   * @return SpringComponentsInfo containing status of GMS, MAE Consumer, and MCE Consumer
+   */
+  public SpringComponentsInfo getSpringComponentsInfo() {
+    return springComponentsCollector.collect(executorService);
+  }
+
+  /**
+   * Get all system properties with detailed metadata.
+   *
+   * <p>Returns comprehensive property information including:
+   *
+   * <ul>
+   *   <li>Individual property details with sources and resolution
+   *   <li>Property source metadata
+   *   <li>Filtering and redaction statistics
+   * </ul>
+   *
+   * @return SystemPropertiesInfo with detailed property metadata
+   */
+  public SystemPropertiesInfo getSystemPropertiesInfo() {
+    return propertiesCollector.collect();
+  }
+
+  /**
+   * Get only configuration properties as a simple map (for backward compatibility).
+   *
+   * <p>This method provides a simplified view of system properties without metadata, suitable for
+   * legacy integrations or simple configuration debugging.
+   *
+   * @return Map of property keys to resolved values
+   */
+  public Map<String, Object> getPropertiesAsMap() {
+    return propertiesCollector.getPropertiesAsMap();
+  }
+
+  /**
+   * Get complete system information - spring components and properties in parallel.
+   *
+   * <p>This is the primary method for retrieving comprehensive system information. It executes both
+   * component and property collection in parallel for optimal performance.
+   *
+   * @return SystemInfoResponse containing both component and property information
+   * @throws SystemInfoException if collection fails or times out
+   */
+  public SystemInfoResponse getSystemInfo() {
+    List<CompletableFuture<?>> futures =
+        Arrays.asList(
+            CompletableFuture.supplyAsync(this::getSpringComponentsInfo, executorService),
+            CompletableFuture.supplyAsync(this::getSystemPropertiesInfo, executorService));
```

Comment:
@david-leifker the structure of the data is copied from the original pr - I find it a little strange that SpringComponents contain data included to SystenProperties (data collected by PropertiesCollector). The data displayed a little different - spring components display it as key value pair, and  SystemProperties displays with additional data (like source, sourceType etc)